library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -994,6 +1001,9 @@ public int getMaxRetryCount()
    */
   public void setMaxRetryCount(int maxRetryCount)
   {
+    if (maxRetryCount < 0) {
+      throw new IllegalArgumentException("maxRetryCount should be greater than or equal to 0.");
+    }
```

Comment:
Are talking about dynamic property changes ? If yes, I think we need to fix it up in core.